src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java

```diff
@@ -872,28 +884,14 @@ public boolean evaluate(Object input) {
         return getById(id, clazz);
     }
 
-    private <T> void updateResourceLayerName(ResourceInfo info) {
-        final Object newValue = info.getName();
-        Filter filter = Predicates.equal("resource.id", info.getId());
-        List<LayerInfo> resourceLayers;
-        resourceLayers = this.queryAsList(LayerInfo.class, filter, null, null, null);
-        for (LayerInfo layer : resourceLayers) {
-            Set<PropertyType> propertyTypes = dbMappings.getPropertyTypes(LayerInfo.class, "name");
-            PropertyType propertyType = propertyTypes.iterator().next();
-            Property changedProperty = new Property(propertyType, newValue);
-            Integer layerOid = findObjectId(layer);
-            updateQueryableProperties(layer, layerOid, ImmutableSet.of(changedProperty));
-        }
-    }
-
-    private <T> void updateResourceLayerKeywords(ResourceInfo info) {
-        final Object newValue = info.getKeywords();
+    private <T> void updateResourceLayerProperty(ResourceInfo info, String propertyPath) {
+        final Object newValue = info.isAdvertised();
```

Comment:
This uses the isAdvertised() value for all of the properties.  Maybe add newValue as a parameter to this method and pass in the correct value?